### PR TITLE
[Do_Not_Merge]Add support to edit config on remote network device

### PR DESCRIPTION
**Pull request by ganeshrn**
Add a generic edit_config module that is independent
underlying transport and can work with all the
supported connection type for network devices (newtork_cli,
netconf, local)

### CODE REVIEW

File: library/edit_config.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    source = module.params['source']
+    config = module.params['config']
+
+    conn = Connection(module._socket_path)
+    capability = json.loads(conn.get_capabilities())
+    result = {'changed': False}
+
+    if module.params['backup']:
+        output = conn.get_config(source=source, format=format)
```
Comment (trishnaguha) [inline library/edit_config.py:103]: `format` is not defined.

File: library/edit_config.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        result['__backup__'] = to_text(output, errors='surrogate_then_replace').strip()
+
```
Comment (trishnaguha) [inline library/edit_config.py:105]: writing backup file functionality is required as this won't hit plugins/action/${platform_specific}_config.py

File: library/edit_config.py (hunk 1 above)
Comment (ganeshrn) [reply library/edit_config.py:103]: Good catch. Fixed it now.

File: library/edit_config.py (hunk 2 above)
Comment (ganeshrn) [reply library/edit_config.py:105]: This is taken care by `edit_config` action plugin, so not required to have platform specific config plugin.

**Review by trishnaguha — APPROVED** [review]

**Review by privateip — CHANGES_REQUESTED** [review]
I would like to have a longer conversation about this module as this implementation is not correct imho.  We should discuss this in irc meeting

**Comment by gundalow** [comment]
recheck

**Comment by ganeshrn** [comment]
Closing this PR, as this functionality will be added as part of `network_config` module.

**Pull request closed without merging**